src/articles/FindArticles.js

```diff
@@ -55,6 +63,42 @@ export default function NewArticles() {
     );
   };
 
+  function handleScroll() {
+    let scrollBarPixelDistToPageEnd = getPixelsFromScrollBarToEnd({});
+    if (
+      scrollBarPixelDistToPageEnd <= 50 &&
+      !isWaitingForNewArticlesRef.current
+    ) {
+      setIsWaitingForNewArticles(true);
+      let newCurrentPage = currentPageRef.current + 1;
+      let newArticles = [...articleListRef.current];
+      insertNewArticlesIntoArticleList(newCurrentPage, newArticles);
+    }
+  }
+
+  function insertNewArticlesIntoArticleList(newCurrentPage, newArticles) {
+    document.title = "Getting more articles...";
+    if (searchQuery) {
+      api.searchMore(searchQuery, newCurrentPage, (articles) => {
+        newArticles = newArticles.concat(articles);
+        setArticleList(newArticles);
+        setOriginalList([...newArticles]);
```

Comment:
there is still duplication that could be factored out between these two conditional branches. or do you prefer not to do that?

Reply:
Oh I misunderstood your comment before about this, yeah no you are right - I have fixed that now!